### PR TITLE
Add support for list service keys

### PR DESCRIPTION
**Pull request by xingzhou**
[#87057920]

### CODE REVIEW

**Comment by maximilien** [comment]
@shalako this PR is pending on CC and was submitted yesterday and adds the listing of service keys API behavior.

**Comment by shalako** [comment]
@maximilien story to review has been prioritized

File: app/controllers/services/service_instances_controller.rb [hunk 1]
```diff
@@ -175,6 +175,33 @@ def permissions(guid)
       end
     end
 
+    get '/v2/service_instances/:guid/service_keys', :service_keys
+    def service_keys(guid)
+      logger.debug 'cc.enumerate.related', guid: guid, association: 'service keys'
```
Comment (maximilien) [inline app/controllers/services/service_instances_controller.rb]: should we include `logger.debug` statements in the production code?

File: app/controllers/services/service_instances_controller.rb [hunk 2]
```diff
@@ -175,6 +175,33 @@ def permissions(guid)
       end
     end
 
+    get '/v2/service_instances/:guid/service_keys', :service_keys
```
Comment (jeffh) [inline app/controllers/services/service_instances_controller.rb]: Hey, @xingzhou, please remove this method and use the `define_attributes` method listed on the top of this file: https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/controllers/services/service_instances_controller.rb#L10-L16 to be consistent with service bindings. `define_attributes` will implement the nested relationship.

Adding `to_many :service_keys` will implement this entire method. Some of the tests that verify the attributes will need to be updated. Also, the one_to_many relationship of service instances to service keys should be added to the [service instance model](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/app/models/services/service_instance.rb#L21) for this to work.

Keep the existing tests that you have written for `/v2/service_instance/:guid/service_keys` endpoint.

Thanks,
Jeff and @rmasand, Services API Team

Comment (xingzhou) [inline app/controllers/services/service_instances_controller.rb]: Hi @jeffh, this is Tom from IBM Beijing lab, thanks for your comments, one question is do we have tests that verify the `define_attributes`? if no, can we just remove the whole method and add the `to_many` relationship and resubmit this PR?

**Comment by xingzhou** [comment]
Hi @jeffh , this is Edward and Tom from IBM Beijing team, thanks for your review. We have resubmitted this PR by refining the code according to your comments. Please take a look, thanks

**Comment by jeffh** [comment]
Thanks @xingzhou. Merged. :+1:

**Pull request merged**